### PR TITLE
Support setting tolerations, nodeSelector, and affinity during helm install

### PR DESCRIPTION
**Pull request by linglingye001**

### CODE REVIEW

File: deploy/templates/autoscaling.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: autoscaling/v2
```
Comment (RichardChen820) [inline deploy/templates/autoscaling.yaml:1]: Shouldn't he HPA only be deployed while the autoscaling is enabled?

File: deploy/templates/autoscaling.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: az-appconfig-k8s-provider-hpa
```
Comment (RichardChen820) [inline deploy/templates/autoscaling.yaml:5]: use the full name variable

File: deploy/parameter/helm-values.yaml [hunk 3]
```diff
@@ -41,18 +41,31 @@ securityContext:
 
 resources: 
   limits:
-    cpu: 500m
+    cpu: 100m
     memory: 128Mi
   requests:
-    cpu: 10m
+    cpu: 20m
```
[the remark this answers is not in the document]
Comment (linglingye001) [reply deploy/parameter/helm-values.yaml:47]: image

File: deploy/parameter/helm-values.yaml [hunk 4]
```diff
@@ -41,18 +41,31 @@ securityContext:
 
 resources: 
   limits:
-    cpu: 500m
+    cpu: 100m
     memory: 128Mi
```
Comment (RichardChen820) [inline deploy/parameter/helm-values.yaml:45]: Seems double the request memory is too conservative. How about using 256Mi as a limit?

Comment (linglingye001) [reply deploy/parameter/helm-values.yaml]: Updated

**Review by RichardChen820 — APPROVED** [review]

**Pull request merged**